handler.go

```diff
@@ -255,10 +312,14 @@ func checkHandleType(t reflect.Type) (*requestType, error) {
 			return nil, errgo.Newf("no argument parameter after Params argument")
 		}
 	}
-	pt, err := getRequestType(t.In(t.NumIn() - 1))
+	argt := t.In(t.NumIn() - 1)
+	pt, err := getRequestType(argt)
 	if err != nil {
 		return nil, errgo.Notef(err, "last argument cannot be used for Unmarshal")
 	}
+	if argInterfacet != nil && !argt.Implements(argInterfacet) {
+		return nil, errgo.Notef(err, "argument does not implement interface required by root handler %v", argInterfacet)
```

Comment:
If possible could this error message mention both types "%v does not implement %v" it would be nice if it told you why but lets not implement a compiler.